apps/settings/js/modules/bluetooth/bluetooth_context.js

```diff
@@ -0,0 +1,809 @@
+     * @param {event} evt
+     */
+    _onAdapterDeviceunpaired:
+    function btc__onAdapterDeviceunpaired(adapter, evt) {
+      Debug('_onAdapterDeviceunpaired evt = ' + evt);
+      // have to get the device address in this event handler Ex. evt -> address
+
+      // Instead of removing the paired device from paired devices list,
+      // get paired devices from adapter directly.
+      this._refreshPairedDevicesInfo(adapter);
+    },
+
+    /**
+     * 'defaultAdapter' change event handler from adapter manager for
+     * watch/unwatch default adapter relative event immediately.
+     *
+     * @access private
+     * @memberOf BluetoothContext
+     * @param {Object BluetoothAdapter} newAdapter
+     * @param {Object BluetoothAdapter} oldAdapter
+     */
+    _onDefaultAdapterChanged:
+    function btc__onDefaultAdapterChanged(newAdapter, oldAdapter) {
+      Debug('_onDefaultAdapterChanged(): newAdapter = ' + newAdapter);
+      Debug('_onDefaultAdapterChanged(): oldAdapter = ' + oldAdapter);
+
+      // save default adapter
+      this._defaultAdapter = newAdapter;
+
+      if (oldAdapter) {
+        // unwatch event since the old adapter is no longer usefull
+        this._unwatchDefaultAdapterOnattributechanged(oldAdapter);
+        this._unwatchDefaultAdapterOndeviceunpaired(oldAdapter);
+        this._unwatchDefaultAdapterOndevicepaired(oldAdapter);
+      }
+
+      if (newAdapter) {
+        // watch event since the new adapter is ready to access
+        this._initProperties(newAdapter);
+        this._watchDefaultAdapterOnattributechanged(newAdapter);
+        this._watchDefaultAdapterOndevicepaired(newAdapter);
+        this._watchDefaultAdapterOndeviceunpaired(newAdapter);
+      } else {
+        // reset properties only
+        this._resetProperties();
+      }
+    },
+
+    /**
+     * Refresh paired devices information.
+     *
+     * @access private
+     * @memberOf BluetoothContext
+     * @param {Object BluetoothAdapter} adapter
+     */
+    _refreshPairedDevicesInfo: function btc__refreshPairedDevicesInfo(adapter) {
+      var pairedDevices = adapter.getPairedDevices();
+      Debug('pairedDevices.length = ' + pairedDevices.length);
+
+      // reset paired devices list
+      this._pairedDevices.reset([]);
+
+      // refresh properties about paired devices
+      if (pairedDevices.length === 0) {
+        // reset the name
+        this.firstPairedDeviceName = '';
+        // update property 'hasPairedDevice'
+        this.hasPairedDevice = false;
+      } else {
+        // save paired devices in list
+        for (var i in pairedDevices) {
+          // create observable BtDevice
+          var observableBtDevice = BtDevice(pairedDevices[i]);
+          // push device in devices list with observable object
+          this._pairedDevices.push(observableBtDevice);
+        }
+
+        // sort paired devices in list
+        this._pairedDevices.array.sort(function(a, b) {
```

Comment:
Note that `ListView` uses `this._pairedDevice.array` to generate the list. This line will make the list sorted by name and I believe this is not what we want.